spikeinterface/toolkit/qualitymetrics/misc_metrics.py

```diff
@@ -139,23 +151,28 @@ def compute_isi_violations(waveform_extractor, isi_threshold_ms=1.5):
     seg_durations = [recording.get_num_samples(i) / fs for i in range(num_segs)]
     total_duraion = np.sum(seg_durations)
 
-    isi_threshold = (isi_threshold_ms / 1000. * fs)
+    isi_threshold_s = isi_threshold_ms / 1000
+    min_isi_s = min_isi_ms / 1000
 
-    isi_violations_count = {}
     isi_violations_rate = {}
 
+    # all units converted to seconds
     for unit_id in unit_ids:
         num_violations = 0
+        num_spikes = 0
         for segment_index in range(num_segs):
-            st = sorting.get_unit_spike_train(unit_id=unit_id, segment_index=segment_index)
-            isi = np.diff(st)
-            num_violations += np.sum(isi < isi_threshold)
+            spike_train_s = sorting.get_unit_spike_train(unit_id=unit_id, segment_index=segment_index) * fs
+            isis = np.diff(spike_train_s)
+            num_spikes += len(spike_train_s)
+            num_violations += sum(isis < isi_threshold_s)
+        violation_time = 2 * num_spikes * (isi_threshold_s - min_isi_s)
+        total_rate = num_spikes / total_duraion
+        violation_rate = num_violations / violation_time
+        isi_rate = violation_rate / total_rate
 
-        isi_violations_count[unit_id] = num_violations
-        isi_violations_rate[unit_id] = num_violations / total_duraion
+        isi_violations_rate[unit_id] = isi_rate
 
-    res = namedtuple('isi_violaion', ['isi_violations_rate', 'isi_violations_count'])
-    return res(isi_violations_rate, isi_violations_count)
+    return isi_violations_rate
```

Comment:
I think it was a good stuff to have both the number and the rate. Why do you remove the cuonting ?

Reply:
I think counting is irrelevant because it depends on duration and it can't be used as a metric

Reply:
I think it is cool to have anyway.
The namedtupple allow to have sevreal columns for the dataframe. So several "metrics" for one function.